src/main/java/com/rackspace/feeds/feedscatalog/ResolveHostFilter.java

```
@@ -0,0 +1,138 @@
+
+        // setup wrapper response with output stream to collect transformed content
+        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        ServletOutputStreamWrapper outputStreamWrapper = new ServletOutputStreamWrapper(stream);
+        OutputStreamResponseWrapper wrappedResponse =
+                new OutputStreamResponseWrapper(httpServletResponse, outputStreamWrapper);
+
+        // apply filter further down the chain on wrapped response
+        chain.doFilter(httpServletRequest, wrappedResponse);
+
+        // obtain response content
+        String originalResponseContent = stream.toString();
+        LOG.debug("Original response content length = " + originalResponseContent.length());
+
+        if (StringUtils.isNotEmpty(originalResponseContent)) {
+
+            // create input params to xslt with headers from request
+            HashMap<String, Object> params = new HashMap<String, Object>();
+            putTenantIdParams(httpServletRequest, params);
+            putExternalLocParams(httpServletRequest, params);
+
+            try {
+                OutputStream outputStream = new ByteArrayOutputStream();
+
+                // transform response content with resolve host xslt
+                transformer.doTransform(params,
+                    new StreamSource(new StringReader(originalResponseContent)),
+                    new StreamResult(outputStream));
+
+                // set transformed content to response
+                String newResponseContent = outputStream.toString();
+                LOG.debug("New response content length = " + newResponseContent.length());
+                httpServletResponse.setContentLength(newResponseContent.length());
+                httpServletResponse.getWriter().write(newResponseContent);
+            }
+            catch (Exception e) {
+                LOG.error("Error transforming xml: " + e.getMessage());
```

If there's an error transforming the xml, we should throw the exception. I.e: just pass the exception to the caller (the servlet engine) so it knows to return 500 error to clients.